source/src/EventSource.cc

```diff
@@ -0,0 +1,287 @@
+        }
+        
+        // if not yet registered to the collector, do it
+        if(!iter->second.m_registered)
+        {
+          // get it once
+          if(sourceInfo.empty())
+            this->getSourceInfo(sourceInfo);
+            
+          iter->second.m_registered = this->registerMe(iter->first, sourceInfo);
+        }
+        
+        if(!iter->second.m_registered)
+        {
+          dqm_warning( "EventSource::sendEvent(): Source '{0}' not registered yet to collector '{1}'. Skipping ...", m_sourceName, collector );
+          continue;
+        }
+        
+        std::string collectCommand = "/dqm4hep/app/evtcol/" + collector + "/collect";
+        m_client.sendCommand(collectCommand, collectBuffer);
+      }  
+    }
+
+    //-------------------------------------------------------------------------------------------------
+
+    void EventSource::getSourceInfo(Json::Value &info)
+    {
+      // source name
+      info["source"] = m_sourceName;
+      
+      // host info
+      StringMap hostInfo; Json::Value hostInfoValue;
+      core::fillHostInfo(hostInfo);
+      
+      for(auto hi : hostInfo)
+        hostInfoValue[hi.first] = hi.second;
+      
+      info["host"] = hostInfoValue;
+      
+      // collectors
+      Json::Value collectorsValue;
+      for(auto colIter : m_collectorInfos)
+        collectorsValue.append(colIter.first);
+      
+      info["collectors"] = collectorsValue;
+      
+      // streamer name
+      info["streamer"] = m_streamerName;
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    bool EventSource::registerMe(const std::string &collector, const Json::Value &info)
+    {
+      std::string requestName = "/dqm4hep/app/evtcol/" + collector + "/register";
+      Json::StreamWriterBuilder builder;
+      std::unique_ptr<Json::StreamWriter> writer(builder.newStreamWriter());
+      std::ostringstream infoStr;
+      writer->write(info, &infoStr);
+      
+      bool returnValue(false);
+      net::Buffer requestBuffer;
+      auto model = requestBuffer.createModel<std::string>();
+      requestBuffer.setModel(model);
+      model->move(std::move(infoStr.str()));
+      
+      m_client.sendRequest(requestName, requestBuffer, [&returnValue,&collector](const net::Buffer &buffer){
+        Json::Value response;
+        Json::CharReaderBuilder readerBuilder;
+        std::unique_ptr<Json::CharReader> reader(readerBuilder.newCharReader());
+        reader->parse(buffer.begin(), buffer.end(), &response, nullptr);
+        
+        const bool registered(response.get("registered", false).asBool());
+        
+        if(!registered)
+        {
+          const std::string message(response.get("message", "The event collector is maybe not available or is down !").asString());
+          dqm_warning( "EventSource::registerMe(): Couldn't register to source to collector '{0}': {1}", collector, message );
```

Comment:
Small typo : Couldn't register `to` source to collector

Reply:
Fixed